docs/2.guide/3.going-further/3.modules.md

```diff
@@ -127,7 +127,19 @@ export default function (inlineOptions, nuxt) {
 }
 ```
 
-Outside of short inline modules defined in `nuxt.config.ts`, **we do not recommend** using this low-level function definition. Instead, to define a module, **we recommend** using the higher-level `defineNuxtModule` helper provided by [Nuxt Kit](/docs/api/advanced/kit).
+You can get type-hint support for this function using the higher-level `defineNuxtModule` helper provided by [Nuxt Kit](/docs/api/advanced/kit).
+
+```js
+import { defineNuxtModule } from '@nuxt/kit'
+
+export default defineNuxtModule((options, nuxt) => {
+  nuxt.hook('pages:extend', pages => {
+    console.log(`Discovered ${pages.length} pages`)
+  })
+})
+```
+
+However, outside of short inline modules defined in `nuxt.config.ts`, **we do not recommend** using this low-level function definition. Instead, to define a module, **we recommend** using the object-syntax with `meta` property to identify your module.
```

Comment:
I think we don't need a helper for inline modules in `nuxt.config` because the types should be automatically inferred when used within the configuration definition.

But you raise a good point here. My main reservation is that by allowing this functional syntax we can miss out on the opportunity to de-duplicate modules (as there is no metadata).

Thoughts welcome from others.

Reply:
Maybe mentioning `nuxt.config.ts` could be a bit confusing as this is the module author guide, while inline-function usage is shown in configuration - https://nuxt.com/docs/api/configuration/nuxt-config#modules

Suggested change:
```suggestion
However, **we do not recommend** using this low-level function definition. Instead, to define a module, **we recommend** using the object-syntax with `meta` property to identify your module.
```

Reply:
Maybe mentioning `nuxt.config.ts` could be a bit confusing as this is the module author guide, while inline-function usage is shown in configuration - https://nuxt.com/docs/api/configuration/nuxt-config#modules

Suggested change:
```suggestion
However, **we do not recommend** using this low-level function definition. Instead, to define a module, **we recommend** using the object-syntax with `meta` property to identify your module.
```